packages/at_lookup/lib/src/at_lookup_impl.dart

```diff
@@ -386,9 +386,10 @@ class AtLookupImpl implements AtLookUp {
     if (privateKey == null) {
       throw UnAuthenticatedException('Private key not passed');
     }
-    await createConnection();
+
     try {
       await _pkamAuthenticationMutex.acquire();
+      await createConnection();
```

Comment:
should the createConnection be before mutex acquire?

Reply:
If the create connection is before the mutex acquire and the createConnection() encounters an exception; it will execute the finally block where we are trying to release a mutex that hasn't been acquired yet. Alternatively, we can put createConnection() in a separate try-catch block where it can return false upon encountering an exception. This way we can have the mutex acquire after the createConnection().